DS implementations/QUEUES/Queue_Implementation.cpp

```diff
@@ -0,0 +1,138 @@
+// C++ program to implement a queue using an array
+
+#include <bits/stdc++.h>
+using namespace std;
+
+class Queue
+{
+    int front, rear, capacity; 
+    int *queue;
+
+public:
+    // constructor to form a queue of capacity c.
+    Queue(int c)
+    {
+        // front wil show the start index and rear will show the end index of the queue and because 
+        // we haven't fill any element in the queue yet so rear and front both are at starting index.
+        front = rear = 0;
+        capacity = c;
+        queue = new int;
+    }
+
+    // destructor to destroy queue.
+    ~Queue() 
+    { 
+        delete[] queue; 
+    }
+
+    // function to insert an element at the rear of the queue
+    void queueEnqueue(int data)
+    {
+        // check queue is full or not
+        if (capacity == rear)
+        {
+            printf("\nQueue is full\n");
+            return;
+        }
+
+        // insert element at the rear
+        else
+        {
+            queue[rear] = data;
+            rear++;
+        }
+        return;
+    }
+
+    // function to delete an element from the front of the queue
+    void queueDequeue()
```

Comment:
The pop operation should be of time complexity `O(1)`, you implemented `O(n)` operation